pkgs/build-support/build-fhsenv-bubblewrap/default.nix

```diff
@@ -187,6 +187,13 @@ let
       x11_args+=(--ro-bind-try "$local_socket" "$local_socket")
     fi
 
+    ${lib.optionalString privateTmp ''
+    # sddm places XAUTHORITY in /tmp
+    if [[ "$XAUTHORITY" == /tmp/* ]]; then
```

Comment:
Should we maybe just always do that? It can't _hurt_.

Reply:
On the other hand, there's no reason *to* do it when `privateTmp` is false, either. This way it's a complete noop for other `buildFHSEnv` packages, rather than a potential source if behavior changes. Just in case someone's doing their own additional binds that it might interfere with, or something.

Reply:
Hmm, I looked at what flatpak is doing, and they're just copying the cookie to a file: https://github.com/flatpak/flatpak/blob/65bc369a9f7851cc1344d2a767b308050cd66fe3/common/flatpak-run-x11.c#L302-L326

Maybe we should also do that?

Reply:
Flatpak's goal is to make a fixed environment as independent as possible from desktop environment and other such concerns, `buildFHSEnv`'s isn't really. The more we make things look like they would with the same software on a normal linux distro, the better. And normal for an sddm-started X11 session is having `XAUTHORITY` in `/tmp`.

Reply:
I just don't like how many different code paths buildFHSEnv is starting to have. I'd rather try to make it more consistent, even if it means a little bit of work that is strictly speaking unnecessary.

Reply:
I'm not sure there's anything to be done about that, really. And without a clear alternate path to take atm, I'd lean toward getting sddm users able to run steam again rather than trying to figure it out right now.

It feels to me like all those code paths are pretty necessary. As much as simplifying would be nice, I just don't see how.